### DISCUSSION TITLE
Could PROXY protocol be enabled per MQTT listener?

### DISCUSSION THREAD

**Discussion by tsigcn** [rabbitmq/rabbitmq-server#12076] [Questions] [answered]
As mentioned before in [#2870](https://github.com/rabbitmq/rabbitmq-server/discussions/2870), PROXY protocol can be enabled per TCP listener, could it be enabled per MQTT listener?
I have tried the following configuration,

mqtt.listeners.tcp.1 = 12345
mqtt.listeners.tcp.1.proxy_protocol = true

But it cannot work, is there another parameter for this?
Any advice will be appreciated.

**Comment by michaelklishin** [comment, marked as answer]
@tsigcn are you sure that as of https://github.com/rabbitmq/rabbitmq-server/discussions/2870 proxy protocol can be enabled per listener? Because I am not aware of such change, and https://github.com/rabbitmq/rabbitmq-server/discussions/2870 seems to be "just an idea thrown out there."

Perhaps look for pull requests that are merged (and were not reverted) and not for discussions if you want to consider if a change was introduced or not.

Proxy protocol is enabled node-wide, as [the docs](https://www.rabbitmq.com/docs/networking#proxy-protocol) and [the source code of `rabbitmq.conf` mappings](https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbit/priv/schema/rabbit.schema#L1575-L1580) suggest.

I haven't tested it with MQTT in a very long time but it should apply to all protocols. Selectively enabling Proxy protocol (supporting it alongside "regular" connections) is considered to be a security flaw in the Proxy protocol spec.
